File: 프로그래머스/1/136798.기사단원의 무기/기사단원의 무기.py
def get_factor_count(n):
    count = 0
    
    for i in range(1, int(n ** 0.5) + 1):
        if n % i == 0:
            count += 1 if i * i == n else 2
            
    return count

def solution(number, limit, power):
    answer = 0
    
    for i in range(1, number + 1):
        cnt = get_factor_count(i)
        
        if cnt > limit:
            answer += power
        else:
            answer += cnt
    
    return answer

File: 프로그래머스/1/136798.기사단원의 무기/기사단원의 무기.java
class Solution {
    public int[] getFactorCnt(int number) {
        int[] returnArr = new int[number];
        
        for (int i = 0; i < number; i++) {
            int factorCnt = 0;
            for (int j = 1; j <= i + 1; j++) {
                if ((i + 1) % j == 0) {
                    factorCnt++;
                }
            }
            returnArr[i] = factorCnt;
        }
        
        return returnArr;
    }
    
    public int solution(int number, int limit, int power) {
        int answer = 0;
        
        
        
        // number의 각 약수 개수 구하기
        int[] factor = new int[number];
        factor = getFactorCnt(number);

        
        // 약수의 개수가 공격력 제한 수치를 넘는지 확인
        for (int i = 0; i < number; i++) {
            if (factor[i] > limit) {
                factor[i] = power;
            }
        }
        
        // 합 구하기
        for (int i = 0; i < number; i++) {
            answer += factor[i];
        }
        
        return answer;
    }
}